File: lcbp_a-frontend/src/app/interfaces/shop.ts
import { Event } from "./event";
import { Product } from "./product";

export interface Shop{
    id_shop:number,
    id_user:number,
    id_shoprequest:number,
    name:string,
    description:string,
    address:string,
    phone:string,
    siret:string,
    logo:string,
    image:string,
    legalproof:string,
    id_status:number,
    products:Product[],
    events:Event[],
    promos:Product[],
}

File: lcbp_a-mobile/lib/entities/shop.dart

import 'dart:ffi';

class Shop {

  final String name;
  final String description;
  final String address;
  final String image;
  final String phone;
  final String siret;
  final String legalproof;
  final String logo;

  const Shop({
    required this.name,
    required this.description,
    required this.address,
    required this.image,
    required this.phone,
    required this.siret,
    required this.legalproof,
    required this.logo
  });

  factory Shop.fromJson(Map<String, dynamic> json) {
    return Shop(
      name: json['name'],
      description: json['description'],
      address: json['address'],
      image: json['image'],
      phone: json['phone'],
      siret: json['siret'],
      legalproof: json['legalproof'],
      logo: json['logo'],
    );
  }
}